<a href="https://colab.research.google.com/github/TrongAdonis/DataAnalystDeepLearning/blob/main/Btvn5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Bài Tập

Bước 1: Tính các đại lượng thống kê cho hai tập dữ liệu

In [1]:
import pandas as pd
dulieu = pd.read_csv('https://raw.githubusercontent.com/huynhhoc/DataAnalystDeepLearning/main/Data/babies.csv', sep='\s+')
dulieu

bwt  smoke
0     120      0
1     113      0
2     128      1
3     123      0
4     108      1
...   ...    ...
1231  113      0
1232  128      0
1233  130      1
1234  125      0
1235  117      0

[1236 rows x 2 columns]

In [2]:
dl1 = pd.DataFrame(dulieu)
dlhutthuoc = dl1[dl1['smoke'] == 1]
dlkohutthuoc = dl1[dl1['smoke'] == 0]

In [3]:
d = {'TH1: Bà mẹ hút thuốc' : pd.Series([dlhutthuoc['bwt'].count(),dlhutthuoc['bwt'].min(),
                        dlhutthuoc['bwt'].max(),dlhutthuoc['bwt'].mean(),
                        dlhutthuoc['bwt'].std(),dlhutthuoc['bwt'].var(),
                        dlhutthuoc['bwt'].median(),dlhutthuoc['bwt'].quantile(.0),
                        dlhutthuoc['bwt'].quantile(.25),dlhutthuoc['bwt'].quantile(.5),
                       dlhutthuoc['bwt'].quantile(.75),dlhutthuoc['bwt'].quantile(1),
                       dlhutthuoc['bwt'].quantile(.75)-dlhutthuoc['bwt'].quantile(.25),
                       dlhutthuoc['bwt'].kurtosis()+2.98,dlhutthuoc['bwt'].skew()],
                       index=['Số lượng','Min','Max','Mean','Sd','Var','Median','Quantile 0%',
                              'Quantile 25%','Quantile 50%','Quantile 75%','Quantile 100%','IQR',
                              'Kurtosis','Skewness']),
     'TH2: Bà mẹ không hút thuốc' : pd.Series([dlkohutthuoc['bwt'].count(),dlkohutthuoc['bwt'].min(),
                        dlkohutthuoc['bwt'].max(),dlkohutthuoc['bwt'].mean(),
                        dlkohutthuoc['bwt'].std(),dlkohutthuoc['bwt'].var(),
                        dlkohutthuoc['bwt'].median(),dlkohutthuoc['bwt'].quantile(.0),
                        dlkohutthuoc['bwt'].quantile(.25),dlkohutthuoc['bwt'].quantile(.5),
                       dlkohutthuoc['bwt'].quantile(.75),dlkohutthuoc['bwt'].quantile(1),
                       dlkohutthuoc['bwt'].quantile(.75)-dlkohutthuoc['bwt'].quantile(.25),
                       dlkohutthuoc['bwt'].kurtosis()+2.98,dlkohutthuoc['bwt'].skew()],
                       index=['Số lượng','Min','Max','Mean','Sd','Var','Median','Quantile 0%',
                              'Quantile 25%','Quantile 50%','Quantile 75%','Quantile 100%','IQR',
                              'Kurtosis','Skewness'])}

In [4]:
dt = pd.DataFrame(d)
dt

TH1: Bà mẹ hút thuốc  TH2: Bà mẹ không hút thuốc
Số lượng                 484.000000                  742.000000
Min                       58.000000                   55.000000
Max                      163.000000                  176.000000
Mean                     114.109504                  123.047170
Sd                        18.098946                   17.398689
Var                      327.571835                  302.714371
Median                   115.000000                  123.000000
Quantile 0%               58.000000                   55.000000
Quantile 25%             102.000000                  113.000000
Quantile 50%             115.000000                  123.000000
Quantile 75%             126.000000                  134.000000
Quantile 100%            163.000000                  176.000000
IQR                       24.000000                   21.000000
Kurtosis                   2.980408                    4.032213
Skewness                  -0.033700                   -0.187363

# Bước 2: Phân tích dữ liệu dựa trên các đại lượng vừa tính
1 Xét tập dữ liệu ứng với trường hợp bà mẹ có hút thuốc
Vị trí tập trung của dữ liệu: khoảng giá trị : 114-115

Tính biến động của dữ liệu:

Phương sai (variance): var=327.571835

Độ lệch chuẩn (standard deviation): sd= 18.098946

Khoảng giá trị: min=58, max=163 -> range=105

Nhận xét: Như vậy dữ liệu phân bố gần nhau.

Hình dạng phân bố của dữ liệu:

Độ lệch: Skewness= -0.033700

Độ bè nhọn của đỉnh dữ liệu: Kurtosis=2.980408

Nhận xét: Như vậy dữ liệu hơi lệch về phía trái, và đỉnh nhọn, hai bên giảm tốc độ vừa phải.


2 Xét tập dữ liệu ứng với trường hợp bà mẹ không hút thuốc

Vị trí tập trung của dữ liệu: khoảng giá trị: 123

Tính biến động của dữ liệu:

Phương sai (variance): 302.714371

Độ lệch chuẩn (standard deviation): sd= 17.398689

Khoảng giá trị: min= 55, max= 176 ->range = 121

Nhận xét: Như vậy dữ liệu hơi lệch về phía trái, và đỉnh nhọn, hai bên giảm tốc độ vừa phải.

Hình dạng phân bố của dữ liệu:

Độ lệch: Skewness= -0.187363

Độ bè nhọn của đỉnh dữ liệu: Kurtosis= 4.032213

Nhận xét: Như vậy dữ liệu hơi lệch về phía trái, và đỉnh nhọn, hai bên giảm tốc độ vừa phải.

Bước 3:So sánh các giá trị thống kê mô tả của hai tập dữ liệu.

Sự khác biệt về vị trí tập trung dữ liệu : chênh lệch khoảng 123-115 = 8

Nhận xét: khác biệt không đáng kể

In [5]:
d1 =pd.DataFrame({'TH1: Bà mẹ hút thuốc' : pd.Series([dlhutthuoc['bwt'].std(skipna=True),dlhutthuoc['bwt'].var(skipna=True),
                                         dlhutthuoc['bwt'].max()-dlhutthuoc['bwt'].min(),
                                         dlhutthuoc['bwt'].quantile(.75)-dlhutthuoc['bwt'].quantile(.25)],
                                         index=['Sd','Var','Range','IQR']),
      'TH2: Bà mẹ không hút thuốc' : pd.Series([dlkohutthuoc['bwt'].std(skipna=True),dlkohutthuoc['bwt'].var(skipna=True),
                                         dlkohutthuoc['bwt'].max()-dlkohutthuoc['bwt'].min(),
                                         dlkohutthuoc['bwt'].quantile(.75)-dlkohutthuoc['bwt'].quantile(.25)],
                                         index=['Sd','Var','Range','IQR'])})
d1['Chênh lệch (TH2-TH1)'] = d1['TH2: Bà mẹ không hút thuốc']-d1['TH1: Bà mẹ hút thuốc']
d1

TH1: Bà mẹ hút thuốc  TH2: Bà mẹ không hút thuốc  Chênh lệch (TH2-TH1)
Sd                18.098946                   17.398689             -0.700257
Var              327.571835                  302.714371            -24.857464
Range            105.000000                  121.000000             16.000000
IQR               24.000000                   21.000000             -3.000000

In [6]:
d2 = pd.DataFrame({'TH1: Bà mẹ hút thuốc' : pd.Series([dlhutthuoc['bwt'].skew(), dlhutthuoc['bwt'].kurtosis()+2.98],
                   index = ['Skewness','Kurtosis']),
                   'TH2: Bà mẹ không hút thuốc' : pd.Series([dlkohutthuoc['bwt'].skew(), dlkohutthuoc['bwt'].kurtosis()+2.98],
                   index = ['Skewness','Kurtosis'])})
d2['Chênh lệch (TH2-TH1)']=d2['TH2: Bà mẹ không hút thuốc']-d2['TH1: Bà mẹ hút thuốc']
d2

TH1: Bà mẹ hút thuốc  ...  Chênh lệch (TH2-TH1)
Skewness             -0.033700  ...             -0.153664
Kurtosis              2.980408  ...              1.051805

[2 rows x 3 columns]